docs/CppInteroperability/GettingStartedWithCppInterop.md

```diff
@@ -0,0 +1,209 @@
+- Navigate to your project directory 
+- In both the `Project` and `Targets`, navigate to `Build Settings` -> `Swift Compiler - Custom Flags`
+- Under `Other Swift Flags` add your search path to the C++ module with the `-I` prefix to import.
+```
+//Add to Other Swift Flags
+-I./ProjectName/Cxx
+```
+
+- This should now allow your to import your C++ Module into any `.swift` file
+```
+//In ViewController.swift
+import UIKit
+import Cxx
+
+class ViewController: UIViewController {
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let result = cxx_function(7)
+        print(result)
+    }
+}
+```
+
+```
+//In Cxx.cpp
+
+#include "Cxx.hpp"
+int cxx_function(int n) {
+    return n;
+}
+
+```
+
+```
+//In Cxx.hpp
+
+#ifndef Cxx_hpp
+#define Cxx_hpp
+
+int cxx_function(int n);
+
+#endif
+
+```
+
+
+## Creating a Swift Package
+- After creating your Swift package project, follow the steps [Creating a Module to contain your C++ source code](#creating-a-module-to-contain-your-c-source-code) in your `Source` directory
+- In your Package Manifest, you need to configure the Swift target's dependencies and compiler flags
+- In this example the name of the package is `Cxx_Interop`
+- Swift code will be in `Sources/Cxx_Interop` called `main.swift`
+- C++ source code follows the example shown in [Creating a Module to contain your C++ source code](#creating-a-module-to-contain-your-c-source-code)
+- Under targets, add the name of your C++ module and the directory containing the Swift code as a target.
+- In the target defining your Swift target, add a`dependencies` to the C++ Module, the `path`, `source`, and `swiftSettings` with `unsafeFlags` with the source to the C++ Module, and enable `-enable-cxx-interop`
+
+```
+//In Package Manifest
+
+import PackageDescription
+
+let package = Package(
+    name: "Cxx_Interop",
+    platforms: [.macOS(.v12)],
+    products: [
+        .library(
+            name: "Cxx",
+            targets: ["Cxx"]),
+        .library(
+            name: "Cxx_Interop",
+            targets: ["Cxx_Interop"]),
+    ],
+    targets: [
+        .target(
+            name: "Cxx",
+            dependencies: []
+        ),
+        //Executable is only needed for using a main.swift, otherwise it can just be a .target
+        .executableTarget(
+            name: "Cxx_Interop",
+            dependencies: ["Cxx"],
+            path: "./Sources/Cxx_Interop",
+            sources: [ "main.swift" ],
+            swiftSettings: [.unsafeFlags([
+                "-I", "Sources/Cxx",
+                "-Xfrontend", "-enable-cxx-interop",
+            ])]
+        ),
+    ]
+)
+
+```
+
+- We are now able to import our C++ Module into our swift code, and import the package into existing projects
+
+```
+//In main.swift
+
+import Foundation
+import Cxx
+
+public struct Cxx_Interop {
+    
+    public func callCxxFunction(n: Int32) -> Int32 {
+        return callCxxFunction(n: n)
+    }
+}
```

Comment:
If this is an executable, maybe add a line at the bottom that prints out the result of `callCxxFunction` or something?